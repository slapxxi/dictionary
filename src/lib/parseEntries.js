// @flow
import _ from 'lodash';
import type { DictionaryEntry } from '../store/types';

type Params = DictionaryEntry;

function parseEntries(
  entries: Array<string>,
  params?: Params,
): Array<DictionaryEntry> {
  return entries.map((entry, index) => {
    const data = entry.split(';');
    return {
      id: index,
      word: data[0].trim(),
      transcription: data[1].trim(),
      definition: data[2].trim(),
      examples: data[3]
        .trim()
        .split(':')
        .map((i) => i.trim())
        .map(_.capitalize),
      thesaurus: data[4].trim().split(','),
      learnt: false,
      viewCount: 0,
      ...(params || {}),
    };
  });
}

export default parseEntries;
