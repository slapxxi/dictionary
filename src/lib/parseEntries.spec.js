import parseEntries from './parseEntries';

it('parses strings to produce dictionary entries', () => {
  const result = parseEntries([
    'word; transcription; definition; example of usage; similar words',
  ]);
  expect(result).toMatchSnapshot();
});

it('merges additional params', () => {
  const result = parseEntries(['w; t; d; usage; similar'], {
    viewCount: 1,
    learnt: true,
  });
  expect(result).toMatchSnapshot();
});
