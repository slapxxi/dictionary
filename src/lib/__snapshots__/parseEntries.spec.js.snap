// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`merges additional params 1`] = `
Array [
  Object {
    "definition": "d",
    "examples": Array [
      "Usage",
    ],
    "id": 0,
    "learnt": true,
    "thesaurus": Array [
      "similar",
    ],
    "transcription": "t",
    "viewCount": 1,
    "word": "w",
  },
]
`;

exports[`parses strings to produce dictionary entries 1`] = `
Array [
  Object {
    "definition": "definition",
    "examples": Array [
      "Example of usage",
    ],
    "id": 0,
    "learnt": false,
    "thesaurus": Array [
      "similar words",
    ],
    "transcription": "transcription",
    "viewCount": 0,
    "word": "word",
  },
]
`;
